### PR TITLE
Begin filtering sensitive data during VCR recording

### PR DESCRIPTION
**Pull request by zleswomp**
Without this change, module testers are left to their own methods to
ensure that they are not leaking any sensitive material when
contributing to this module.  This is not good, as a) we want more tests
and b) we would like not to expose data from contributor environments
when submitting changes.

Here we take a stab addressing this concern by configuring VCR to filter
out the AWS_ACCESS_KEY_ID.  This alone is not enough to be complete, but
it begins the road of making testing simpler on the part of those who
wish to contribute to the module.  Now we can amend the documentation to
include information about exporting the necessary environment variables,
not providers specs don't have to modified last minute to avoid
committing credentials.

### CODE REVIEW

**Comment by zleswomp** [comment]
https://github.com/puppetlabs/puppetlabs-aws/issues/333

**Comment by zleswomp** [comment]
I believe these tests should pass after https://github.com/puppetlabs/puppetlabs-aws/pull/397 is merged and a rebase is done here.

**Comment by hunner** [comment]
Rebase away!

**Comment by zleswomp** [comment]
Rebased and updated the tests for the recently merged iam_ types and providers.

File: lib/puppet/provider/ecs_task_definition/v2.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require_relative '../../../puppet_x/puppetlabs/aws.rb'
+require 'pp'
```
Comment (zleswomp) [inline lib/puppet/provider/ecs_task_definition/v2.rb]: Needs removal.

**Pull request closed without merging**